src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -555,16 +598,18 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
   if(!image.empty()) {
     //image is a sub-option of update
     if(update.empty())
-      throw xrt_core::error("Usage: xbmgmt program --device='0000:00:00.0' --update --image='/path/to/flash_image'");
+      throw xrt_core::error("Usage: xbmgmt program --device='0000:00:00.0' --update --image='/path/to/flash_image' OR shell_name");
     //allow only 1 device to be manually flashed at a time
     if(deviceCollection.size() != 1)
       throw xrt_core::error("Please specify a single device to be flashed");
     //we support only 2 flash images atm
     if(image.size() > 2)
       throw xrt_core::error("Please specify either 1 or 2 flash images");
+    //find the absolute path to the specified image
+    auto image_paths = find_flash_image_paths(image);
     if(!XBU::can_proceed())
       return;
-    update_shell(deviceCollection.front()->get_device_id(), flashType, image.front(), (image.size() == 2 ? image[1]: ""));
+    update_shell(deviceCollection.front()->get_device_id(), flashType, image_paths.front(), (image_paths.size() == 2 ? image_paths[1]: ""));
```

Comment:
Magic Numbers...could you a comment or somehow indicate what is so special about 2.

Reply:
this 2 makes sense when you read the whole function. we support either 1 or 2 flash images. if the user passes in 2 images, then we pass in the second image as well, else we pass in an empty string